File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractspushpushoptionchoice.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CodatDataContractsPushPushOption } from "./codatdatacontractspushpushoption";
import { CodatDataContractsPushOptionTypeEnum } from "./codatdatacontractspushoptiontypeenum";
import { CodatDataContractsPushPushValidationInfo } from "./codatdatacontractspushpushvalidationinfo";



export class CodatDataContractsPushPushOptionChoice extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=description" })
  description: string;

  @SpeakeasyMetadata({ data: "json, name=displayName" })
  displayName: string;

  @SpeakeasyMetadata({ data: "json, name=options", elemType: CodatDataContractsPushPushOptionChoice })
  options?: CodatDataContractsPushPushOptionChoice[];

  @SpeakeasyMetadata({ data: "json, name=properties", elemType: CodatDataContractsPushPushOption })
  properties?: Record<string, CodatDataContractsPushPushOption>;

  @SpeakeasyMetadata({ data: "json, name=rel" })
  rel?: string;

  @SpeakeasyMetadata({ data: "json, name=required" })
  required: boolean;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: CodatDataContractsPushOptionTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=validation" })
  validation?: CodatDataContractsPushPushValidationInfo;

  @SpeakeasyMetadata({ data: "json, name=value" })
  value: string;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_push_pushoptionchoice.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_push_optiontype_enum as shared_codat_datacontracts_push_optiontype_enum
from ..shared import codat_datacontracts_push_pushoption as shared_codat_datacontracts_push_pushoption
from ..shared import codat_datacontracts_push_pushvalidationinfo as shared_codat_datacontracts_push_pushvalidationinfo
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsPushPushOptionChoice:
    description: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    display_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('displayName') }})
    required: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('required') }})
    type: shared_codat_datacontracts_push_optiontype_enum.CodatDataContractsPushOptionTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    value: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('value') }})
    options: Optional[list[CodatDataContractsPushPushOptionChoice]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('options'), 'exclude': lambda f: f is None }})
    properties: Optional[dict[str, shared_codat_datacontracts_push_pushoption.CodatDataContractsPushPushOption]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('properties'), 'exclude': lambda f: f is None }})
    rel: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('rel'), 'exclude': lambda f: f is None }})
    validation: Optional[shared_codat_datacontracts_push_pushvalidationinfo.CodatDataContractsPushPushValidationInfo] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('validation'), 'exclude': lambda f: f is None }})
    